src/MainNav/config.ts

```diff
@@ -108,6 +108,53 @@ export const MainNav: GlobalConfig = {
         },
       ],
     },
+    {
+      name: 'socialLinks',
+      label: 'Social Links',
+      type: 'array',
+      fields: [
+        {
+          name: 'platform',
+          type: 'select',
+          required: true,
+          options: [
+            {
+              label: 'Instagram',
+              value: 'instagram',
+            },
+            {
+              label: 'Facebook',
+              value: 'facebook',
+            },
+            {
+              label: 'Github',
+              value: 'github',
+            },
+          ],
+        },
+        {
+          name: 'link',
+          type: 'text',
+          label: 'URL',
+          required: true,
+          validate: (value: string | string[] | null | undefined) => {
+            if (typeof value !== 'string') {
+              return 'Invalid URL';
+            }
+            const urlPattern = new RegExp(
+              '^(https?:\\/\\/)?' + // protocol
+                '((([a-z\\d]([a-z\\d-]*[a-z\\d])*)\\.?)+[a-z]{2,}|' + // domain name
+                '((\\d{1,3}\\.){3}\\d{1,3}))' + // OR ip (v4) address
+                '(\\:\\d+)?(\\/[-a-z\\d%_.~+]*)*' + // port and path
+                '(\\?[;&a-z\\d%_.~+=-]*)?' + // query string
+                '(\\#[-a-z\\d_]*)?$',
+              'i',
+            ); // fragment locator
+            return !!urlPattern.test(value) || 'Invalid URL';
```

Comment:
Both Node and browsers have a `URL` class that can be used to parse URL strings, and which throws an exception if the string is not a valid URL. I'm pretty sure this could have been written like this (but I haven't tested this code exactly, so I may have made some small mistakes below):

```suggestion
            try {
              const url = new URL(value);
              return !!url;
            } catch (err) {
              return 'Invalid URL';
            }
```

Reply:
I took your suggestion but had to edit slightly in order to be typesafe. Or in the case of err I had to add an underscore before. Don't know if that is the best to do here.

Reply:
I think that's fine! Confirmed fixed in 7597fe9f84d0d211a42532c94c86a8604d488a1c.